Presentation.Web/app/services/exportGridToExcelService.ts

```diff
@@ -91,6 +76,44 @@
             this.columnsToShow = [];
         }
 
+        private selectColumnsToDisplay(e: IKendoGridExcelExportEvent<any>, columns: IKendoGridColumn<any>[], exportOnlyVisibleColumns: boolean) {
+            if (!exportOnlyVisibleColumns) {
+                this.setAllNonExcelOnlyColumnsAsVisible(e, columns);
+            }
+            _.forEach(columns, (column) => {
+                if (!column.hidden) {
+                    this.columnsToShow.push({ columnId: column.persistId });
+                    return;
+                }
+                if (column.parentId === undefined)
+                    return;
+                //look for a parent column and check if any was found, and if any parent column should be visible
+                var columnsWithMatchingParentId = columns.filter(x => x.persistId === column.parentId);
+                if (columnsWithMatchingParentId.length !== 1) {
+                    console.error("Column has multiple columns with matching parentId");
+                    return;
+                }
+                const parentColumn = columnsWithMatchingParentId[0];
+                if (parentColumn.hidden)
+                    return;
+
+                var parentIndex = columns.indexOf(parentColumn);
+
+                this.columnsToShow.push({ columnId: column.persistId, parentId: column.parentId });
+                e.sender.reorderColumn(parentIndex + 1, column);
+            });
+        }
+
+        private setAllNonExcelOnlyColumnsAsVisible(e: IKendoGridExcelExportEvent<any>, columns: IKendoGridColumn<any>[]) {
```

Comment:
since it actually only cares about root columns then maybe that should be the name of the method:
`showAllRootColumns`